LiteCore/tests/VectorQueryTest.cc

```diff
@@ -227,23 +227,160 @@ N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Hybrid Vector Query", "[Query][.Vec
+                 " ORDER BY distance";
+    query                                         = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    std::vector<std::pair<string, float>> result0 = collectResults(query->createEnumerator(&options));
+
+    // Same effect as test 1.
+    CHECK(result0 == result);
+
+
+    // Case3: plain query. If not specified, VECTOR_MATCH will use 3 as max_results.
+    // ==================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target) "
+                 " ORDER BY distance";
+    query  = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    result = collectResults(query->createEnumerator(&options));
+
+    // Argument max_results to VECTOR_MATCH is set by defaut to 3.
+    CHECK(result.size() == 3);
+
+
+    // Case 4: plain query fails to select docs because of the additional WHERE term.
+    // ===================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target, 10) AND META().sequence % 100 = 0"
+                 " ORDER BY distance";
+    query  = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    result = collectResults(query->createEnumerator(&options));
+
+    // This is plain query because of explicit max_results.
+    // VECTOR_MATCH finds first 10 docs without regard to the secondary WHERE term, META().sequence % 100 = 0,
+    // and they are filtered out by that predicate.
+    // Note: hybrid query gives the desired result, c.f. test "Hybrid Vector Query",
+    CHECK(result.size() == 0);
+
+
+    // Asserting that 100 documents are selected by "META().sequence % 100 = 0"
+    queryStr                    = "SELECT count(*) FROM "s + collectionName + " WHERE META().sequence % 100 = 0";
+    query                       = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    Retained<QueryEnumerator> e = query->createEnumerator();
+    REQUIRE(e->next());
+    auto countBySecondaryWhere = e->columns()[0]->asInt();
+    REQUIRE(countBySecondaryWhere == 100);
+
+    // Case 5: hybrid search may not select all the docs that pass the secondary WHERE term.
+    // =====================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target) AND META().sequence % 100 = 0"
+                 " ORDER BY distance";
+    query  = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    result = collectResults(query->createEnumerator(&options));
+
+    // Although hybrid query searches the solutions from the docs that
+    // must pass the secondary WHERE term, it may not select all of them.
+    // In this case, 100 docs passed the predicate META().sequence % 100 = 0, but only 34 are
+    // picked up by the vector query.
+    CHECK(result.size() == 34);
```

Comment:
Don't test for an exact number! That's an implementation detail in vectorsearch. There's no guarantee it might not change if I alter internal algorithms, or even if SQLite's query engine changes slightly.

Reply:
Got it. Will remove it. Maybe it belongs to tutorial or so.